### PR TITLE
Fix: Post button to enable when comments are pasted

### PR DESCRIPTION
**Pull request by lorenjohnson**

### CODE REVIEW

File: src/components/CommentForm.js [hunk 1]
```diff
@@ -102,58 +102,67 @@ export default class CommentForm extends React.PureComponent {
...
+    const { currentUser, isMobile } = this.context
     const { enabled, modifierKey } = this.state
+    const editing = text !== undefined
+    const edit = () => this.setText('')
+    const placeholder = this.props.placeholder || 'Add a comment...'
 
     return <form onSubmit={this.submit} className='comment-form'>
       <Avatar person={currentUser} />
       {editing
         ? <div className='content'>
```
Comment (levity) [inline src/components/CommentForm.js:149]: i guess we're succumbing to the linter on this one ¯\\\_(ツ)_/¯

File: src/components/CommentForm.js [hunk 2]
```diff
@@ -102,58 +102,67 @@ export default class CommentForm extends React.PureComponent {
...
+    const { postId, newComment } = this.props
+    if (!newComment) return
+    if (this.socket) this.socket.post(socketUrl(`/noo/post/${postId}/typing`), { isTyping: false })
+  }
+
+  stopTyping = debounce(this.stoppedTyping, STOPPED_TYPING_WAIT_TIME)
+
+  delaySetText = (text) => {
+    this.setEnabled(text)
+    return debounce(text => this.setText(text), 50)
```
Comment (levity) [inline src/components/CommentForm.js]: `debounce` won't work correctly like this. 

1. you have to assign it to something long-lived and reuse that reference
2. you're not calling the debounced function

you could do this:

```js
if (!this.debouncedSetText) {
  this.debouncedSetText = text => this.setText(text), 50)
}
this.debouncedSetText(text)
```

File: src/components/CommentForm.js (hunk 1 above)
Comment (lorenjohnson) [reply src/components/CommentForm.js:149]: Uh yeah-- I am slowly capitulating and this is a weird case (though I don't disagree with it here). I have my sights on getting us on a CircleCI setup that runs a linter pass eventually and I generally am bringing files I touch for a task (vs a refactor) down to 0 Errors...

File: src/components/CommentForm.js (hunk 2 above)
Comment (lorenjohnson) [reply src/components/CommentForm.js]: Duh. Yeah. I know. I kinda had a thing in the back of mind itching at me as I did it. We just went through this on another component last month. 
I'll back it up, thanks for catching this thing (again)  :)

Comment (lorenjohnson) [reply src/components/CommentForm.js]: So I confirmed that a function assigned as a class property does persist across renders. How I had it was still not right though and made a lot more sense to put the setEnabled within the debounce function either way.

Comment (levity) [reply src/components/CommentForm.js]: Oh, it definitely does. That's why `foo = debounce(() => { ... })` works. But that's not the same as `foo = () => { return debounce(...) }`

Comment (lorenjohnson) [reply src/components/CommentForm.js]: Yup, I get it. Clear enough. Thank you!

File: src/components/CommentForm.js [hunk 3]
```diff
@@ -124,17 +124,17 @@ export default class CommentForm extends React.PureComponent {
...
   setEnabled = (text) => {
     this.setState({enabled: text.length > 0})
   }
 
+  stopTyping = debounce(this.stoppedTyping, STOPPED_TYPING_WAIT_TIME)
+
+  delaySetText = debounce(text => {
+    this.setEnabled(text)
+    this.setText(text)
+  }, 50)
```
Comment (lorenjohnson) [inline src/components/CommentForm.js]: So the above is how it is now.

**Pull request merged**